### PR TITLE
Popup middleware for Orders actions

### PR DESCRIPTION
**Pull request by Velenir**
Middleware that dispatches `Popup` actions on relevant `Orders` actions
`expireOrdersBatch` ation to use instead of individual `expireAction`s in a loop
Removed `addPopup` where middleware takes care of it

https://user-images.githubusercontent.com/5121491/102870231-cd272f80-444d-11eb-9932-3a704fdb31d2.mp4

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr89--gpv2swap.review.gnosisdev.com)**: Testing web app

File: src/custom/state/index.ts [hunk 1]
```diff
@@ -27,22 +28,27 @@ const UNISWAP_REDUCERS = {
...
 const store = configureStore({
-  reducer: {
-    ...UNISWAP_REDUCERS,
-    operator,
-    orders,
-    fee
-  },
-  middleware: [...getDefaultMiddleware({ thunk: false }), save({ states: PERSISTED_KEYS })],
+  reducer: reducers,
+  middleware: [...getDefaultMiddleware({ thunk: false }), save({ states: PERSISTED_KEYS }), popupMiddleware],
```
Comment (anxolin) [inline src/custom/state/index.ts:42]: nit:  I would add some new lines to see the list of middlewares as a vertical list (easier to read)

File: src/custom/state/orders/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    const { pending, fulfilled, expired } = orders
+
+    const orderObject = pending?.[id] || fulfilled?.[id] || expired?.[id]
+
+    // look up Order.summary for Popup
+    const summary = orderObject?.order.summary
+
+    let popup: PopupPayload
+
+    if (isPendingOrderAction(action)) {
```
Comment (anxolin) [inline src/custom/state/orders/middleware.ts:53]: This IF is huge, also the middleware code. Any chance of breaking this up, maybe aux functions?

**Review by anxolin — APPROVED** [review]
Great

**Review by W3stside — APPROVED** [review]

**Comment by W3stside** [comment]
@anxolin regarding this PR as it's Dima's I can take over and finish anything. Pretty familiar with the code as it's middlewares (my fav)

**Comment by anxolin** [comment]
> @anxolin regarding this PR as it's Dima's I can take over and finish anything. Pretty familiar with the code as it's middlewares (my fav)

I appreciate. Thanks

File: src/custom/state/index.ts (hunk 1 above)
Comment (W3stside) [reply src/custom/state/index.ts:42]: this change requires ignoring prettier so will ignore unless u feel strongly about it

**Comment by anxolin** [comment]
@W3stside I think this has the approvals. Can we merge?

**Comment by W3stside** [comment]
@mergify merge

**Pull request merged**